modules/match.js

```diff
@@ -39,18 +39,24 @@ function match({ history, routes, location, ...options }, callback) {
     })
   }
 
-  const router = createRouterObject(history, transitionManager)
-
   transitionManager.match(location, function (error, redirectLocation, nextState) {
-    callback(
-      error,
-      redirectLocation,
-      nextState && {
+    let renderProps
+
+    if (nextState) {
+      const router = {
+        ...createRouterObject(history, transitionManager),
```

Comment:
Can we make `nextState` an arg to `createRouterObject`? And just do `createRouterObject(history, transitionManager, nextState)` here and `createRouterObject(history, transitionManager, this.state)` in `<Router>`?